### ISSUE TITLE
Maestro selector is confused between accessibilityLabel and Text

### ISSUE THREAD

**Issue by Rohit3523** [mobile-dev-inc/maestro-docs#161]
In React Native, we are using `accessibilityLabel` on some components, and in certain cases we pass message content from an API response directly, without any parsing logic. When we run a test using `assertVisible` with a text query, it returns `true` if the `accessibilityLabel` matches the provided text.

Example:
```jsx
<View accessibilityLabel={'#channel'}>
    <Text>channel</Text>
</View>
```

Test:
```yaml
- extendedWaitUntil:
    visible:
      text: '#channel'
    timeout: 10000
```
This test passes, but it *should* fail because the text `#channel` is not actually visible on the screen. Only the accessibility label contains that value.

We also have another case where a message uses `":("` as its `accessibilityLabel`. When we run the following test:
```yaml
- extendedWaitUntil:
    notVisible:
      text: ':('
    timeout: 60000
```

The test returns `false`, even though `":("` is **not** visible on the screen. It seems to fail because the value is present only in the accessibility label, not in any rendered text. The visibility check is incorrectly treating accessibility labels as visible text. I’ve attached a screenshot showing this behavior.

<img width="1512" height="775" alt="Image" src="https://github.com/user-attachments/assets/1f3a1bf3-c15f-43da-a7e1-982455b56c21" />

**Comment by Rohit3523** [comment]
So how can I only match with the text that is currently visible in the screen?

**Comment by Fishbowler** [comment]
Exclusively looking for visible text isn't something that's available right now. You might need to be more creative with your selectors if you've got some overlap between visible text on invisible elements and invisible text in visible elements 😂

Of course, you're also in control of the app under test, so rewording might also be an option, for enhancing testability.
